lib/OptimizationMOI/src/nlp.jl

```diff
@@ -105,7 +105,11 @@ function MOIOptimizationNLPCache(prob::OptimizationProblem, opt; kwargs...)
     reinit_cache = Optimization.ReInitCache(prob.u0, prob.p) # everything that can be changed via `reinit`
 
     num_cons = prob.ucons === nothing ? 0 : length(prob.ucons)
-    f = Optimization.instantiate_function(prob.f, reinit_cache, prob.f.adtype, num_cons)
+    f = Optimization.instantiate_function(prob.f,
```

Comment:
The `ReInitiCache` dispatch of `instantiate_function` doesn't take `p` as an arg https://github.com/SciML/Optimization.jl/blob/ed32f42862b13465bf9fc5ee72c153710b00585a/ext/OptimizationMTKExt.jl#L49-L50.

Reply:
I see, I appear to have confused the two somehow. The idea is to `reinit` the cache rather than the problem then?

Reply:
Yeah https://github.com/SciML/Optimization.jl/issues/500#issuecomment-1499487051 iiuc this is your use case as well? The issue confused me a little as to what you want to do so I might be wrong... 😅